x-pack/plugins/security_solution/server/lib/detection_engine/signals/enrichments/enrichment_by_type/host_risk.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { set, cloneDeep } from 'lodash';
+
+import { getHostRiskIndex } from '../../../../../../common/search_strategy/security_solution/risk_score/common';
+import { RiskScoreFields } from '../../../../../../common/search_strategy/security_solution/risk_score/all';
+import { createSingleFieldMatchEnrichment } from '../create_single_field_match_enrichment';
+import type { CreateRiskEnrichment, GetIsRiskScoreAvailable } from '../types';
+import { getFieldValue } from '../utils/events';
+
+export const getIsHostRiskScoreAvailable: GetIsRiskScoreAvailable = async ({
+  spaceId,
+  services,
+}) => {
+  const isHostRiskScoreIndexExist = await services.scopedClusterClient.asCurrentUser.indices.exists(
+    {
+      index: getHostRiskIndex(spaceId),
+    }
+  );
+
+  return isHostRiskScoreIndexExist;
+};
+
+export const createHostRiskEnrichments: CreateRiskEnrichment = async ({
+  services,
+  logger,
+  events,
+  spaceId,
+}) => {
+  return createSingleFieldMatchEnrichment({
+    name: 'Host Risk',
+    index: [getHostRiskIndex(spaceId)],
+    services,
+    logger,
+    events,
+    mappingField: {
+      eventField: 'host.name',
+      enrichmentField: RiskScoreFields.hostName,
+    },
+    enrichmentResponseFields: [
+      RiskScoreFields.hostName,
+      RiskScoreFields.hostRisk,
+      RiskScoreFields.hostRiskScore,
+    ],
+    createEnrichmentFunction: (enrichment) => (event) => {
```

Comment:
Hey, I think it also makes sense.
But because in the future we will have more enrichments, and we don't know what logic it will have - a function it's a more flexible way.

Maybe the future, when we will know more about enrichment, we can refactor it to the more simple structure